### ISSUE TITLE
Jersey `FormDataMultiPart`

### ISSUE THREAD

**Issue by vguna** [stoicflame/enunciate#569]
Hi.

Is there currently support for uploading binaries like using Jersey's `FormDataMultiPart` using the swagger UI?

I annotated my JAX-RS REST method like so:

```java
	@PUT
	@Consumes(MediaType.MULTIPART_FORM_DATA)
	@Produces(MediaType.APPLICATION_JSON)
	ImageDto upload(FormDataMultiPart image);
```

Which ends up in swagger like so:

![grafik](https://cloud.githubusercontent.com/assets/3257311/21296001/4e246046-c561-11e6-978d-bbf077fc8e68.png)

I would have expected a "select file" button or similar :). AFAIK the topic multipart upload isn't really covered by the JAX-RS spec. So I could understand if this is not supported...

**Comment by stoicflame** [comment]
I'm seeing an upload button.

![issue-569](https://cloud.githubusercontent.com/assets/145838/21322866/3c68fcb4-c5d8-11e6-9ce6-cc4d68985a8f.png)

I basically pasted your code and changed the return type:

```java
  @PUT
  @Consumes (MediaType.MULTIPART_FORM_DATA)
  @Produces ( MediaType.APPLICATION_JSON)
  public Person upload(FormDataMultiPart image) {
    return null;
  }
```

**Comment by vguna** [comment]
I've found some time looking into this again. I've overlooked an additional method definition on the interface:

```java
	@PUT
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	ImageDto update(ImageDto image);
```

The problem here is, that it is the same HTTP method `@PUT` on the same `@Path`, but with a different content-type (JSON vs. MultiPat). Hence I guess the last one simply wins then.

The usecase is, that it is possible to upload a binary file via MultiPart, as well as updating its metadata (e.g. name, description) using the same endpoint. Just by switching the content-type.

For testing I've put the JSON method to another `@Path` and voila, there is the file select button :).

I'm unsure whether swagger even supports this. It offers the content-type selection though. But I don't know what happens if there're two content-types to select from. Does it switch between file/textbox then?

Sorry for commenting to this closed issue,
